source/localizable/tutorial/subroutes.md

```diff
@@ -0,0 +1,425 @@
+
+```shell
+ember g test-helper setup-mirage-for-unit-test
+```
+
+The command creates a test helper file in your `tests/helpers` directory.
+Test helpers are utilities that can be shared between test cases.
+Update the file contents as shown below:
+
+```tests/helpers/setup-mirage-for-unit-test.js
+import mirageInitializer from '../../initializers/ember-cli-mirage';
+
+export default function startMirage(container) {
+  mirageInitializer.initialize(container);
+}
+```
+
+Now let's edit the unit test that Ember created for us when we generated our `show` route.
+
+```tests/unit/routes/rentals/show-test.js
+import { moduleFor, test } from 'ember-qunit';
+import startMirage from '../../../helpers/setup-mirage-for-unit-test';
+import Ember from 'ember';
+
+moduleFor('route:rentals/show', 'Unit | Route | rentals/show', {
+  needs: ['model:rental'],
+  beforeEach() {
+    startMirage(this.container);
+  },
+  afterEach() {
+    window.server.shutdown();
+  }
+});
+
+test('should load rental by id', function(assert) {
+  let route = this.subject();
+  Ember.run(() => {
+    route.model({ rental_id: 'grand-old-mansion'}).then((result) => {
+      assert.equal(result.get('title'), "Grand Old Mansion");
+    });
+  });
+});
+```
+
+In the above unit test we are directly calling the `model` hook method with our rental_id as an argument.
+We then process the resulting promise by asserting that the result has the appropriate title, validating that the correct record was returned.
+
+Also note that in the `beforeEach` hook, we are starting Mirage using the helper we just created, and in the `afterEach` hook we shut down the mirage instance created during start.
+
+## Adding the Rental To Our Template
+
+Next, we can update the template for our show route (`app/templates/rentals/show.hbs`) and list the information for our rental.
+
+```app/templates/rentals/show.hbs
+<div class="jumbo">
+  <a href="#" class="button right up-11">Edit</a>
+  <h2 style="margin-bottom:15px">{{model.title}}</h2>
+  <div class="right" style="width:50%;padding-left:30px">
+    <div class="map right" style="background: url({{model.address}}) center center;"></div>
+    <div class="detail" style="margin-top:10px">
+      <strong>Owner:</strong> {{model.owner}}
+    </div>
+    <div class="detail">
+      <strong>Type:</strong> {{rental-property-type model.type}} - {{model.type}}
+    </div>
+    <div class="detail">
+      <strong>Location:</strong> {{model.city}}
+    </div>
+    <div class="detail">
+      <strong>Number of bedrooms:</strong> {{model.bedrooms}}
+    </div>
+    <p>&nbsp;</p>
+    <p>{{nmodel.description}}</p>
+  </div>
+  <img src="{{model.image}}" alt="" class="" style="width:50%;height:initial;position:static">
+</div>
+```
+
+We pass our `model` to the `rental-listing` component and add a new section listing the rental's `title`.
+Browse to `localhost:4200/rentals/grand-old-mansion` and you should see the information listed for that specific rental.
```

Comment:
I'm not seeing a `rental-listing` component here, do we mean the `rental-property-type` component?